os/include/tinyara/binary_manager.h

```diff
@@ -149,6 +149,9 @@ struct user_binary_header_s {
 	uint32_t bin_ramsize;
 	uint32_t bin_stacksize;
 	uint32_t kernel_ver;
+#ifdef CONFIG_XIP_ELF
```

Comment:
why do you need padding?  If we strictly required word aligned memory for this struct then can we remove the  __attribute__((__packed__)); instead of manually adding padding ourself which might create issues if incase we add some more members in near future.

Reply:
The pc value should be 32 bit aligned, so if we add headers in front of the binary, it wont align without the current padding. I added some comments in mkbinheader.py file in the PR regarding this.